### DISCUSSION TITLE
Web / Cloud OS Adoption

### DISCUSSION THREAD

**Discussion by amegahed** [os-js/OS.js#848] [General]
Hi.  First, congrats on an impressive project.  

I've been following this topic of web or cloud based operating systems in general for about a decade and it seems like these systems have had a very difficult time getting any kind of traction in term of adoption.   With the sole exception of ChromeOS, it's hard to point to examples of other web based OS's that have succeeded.  The space seems to be replete with once promising but failed projects (JoliOS, EyeOS, WebOS etc.)

I was curious if you have theories as to why this is?  A few possibilities:

1) Unfamiliarity
Deploying a cloud OS or building apps on top of a cloud OS is a new and unfamiliar process.  Web developers are familiar with building on top of frameworks, but not web based operating systems.

2) Vendor lock-in
In the US, many areas of tech are dominated by a small number of tech giants which makes it very difficult for new approaches to succeed.   It seems like many web / cloud OS projects are European based, which seems to reflect the vendor lock-in problem.   Could the lack of adoption be because developers are waiting for platforms that are supported by giant tech companies?

3) Risk aversion
For a developer to build upon a base platform, they need to have trust that it will have some amount of longevity.   Given the track record of many of these projects, I can understand the reluctance of developers to take this risk, even if the code is open source.  This is the usual chicken and egg problem that new approaches and technologies face.

4) Ease of use / deployment
Perhaps the process of using / deploying these systems is too great for an uncertain payoff?   Are there things that can be done to make this process easier that would make a difference in adoption?

None of these problems have easy solutions, but I thought that understanding the problem first might help to come up with possible solutions.

**Comment by andersevenrud** [comment]
> it seems like these systems have had a very difficult time getting any kind of traction in term of adoption

I just think it all comes down to that web-based desktop/app-launcher platforms with "traditional" UIs is kind of a niche thing. 

Especially nowadays when it's so simple to deploy applications individually (for any platform) and that remote desktop *streaming* has become super efficient.

**Reply by amegahed** [reply]
Hi Anders!   Thank you for writing.

I think perhaps I'm thinking of os.js a bit differently than it was intended.   I was thinking of os.js not necessarily as a remote desktop replacement, but more as a base platform that you could build apps on top of.

It seems like most web applications today involve aspects that a cloud OS can provide which would make building web applications on top of a cloud OS much easier than trying to tie together a bunch of frameworks like we do today.  

Full disclosure, I'm a developer of a similar cloud OS project (www.sharedigm.com) and I've used it as a base platform to build a number of cloud apps (www.opticexplorer.com, www.dreamachines.ai, etc.)   An an example, the "dreamachines.ai" AI image generation platform was developed in just a few weeks rather than months because the base platform (Sharedigm) already has the support for managing and viewing images, file management, window management, file sharing, etc. 

I imagine a world where a developer might select a cloud OS to build on top of in a similar way that you currently select a front end framework and you would get the whole bundle of support that comes along with it the way you do when you build off of a desktop or mobile OS.   

I see almost no interest in the types of platforms that you and I are building and I was just trying to understand if there's a fundamental reason why or if it's just such an unknown niche that people don't know what these platforms have to offer.
